simulariumio/converter.py

```diff
@@ -0,0 +1,180 @@
+            Loaded data or path to data from a simulation engine. 
+            Fields for each engine:
+
+                custom: 
+                    box_size: np.ndarray (shape = [3])
+                        A numpy ndarray containing the XYZ dimensions 
+                        of the simulation bounding volume
+                    times: np.ndarray (shape = [timesteps])
+                        A numpy ndarray containing the elapsed simulated time 
+                        at each timestep
+                    n_agents: np.ndarray (shape = [timesteps])
+                        A numpy ndarray containing the number of agents 
+                        that exist at each timestep
+                    viz_types : np.ndarray (shape = [timesteps, agents])
+                        A numpy ndarray containing the viz type 
+                        for each agent at each timestep. Current options:
+                            1000 : default,
+                            1001 : fiber (which will require subpoints)
+                    positions : np.ndarray (shape = [timesteps, agents, 3])
+                        A numpy ndarray containing the XYZ position 
+                        for each agent at each timestep
+                    types: List[List[str]] (list of shape [timesteps, agents])
+                        A list containing timesteps, for each a list of 
+                        the string name for the type of each agent
+                    radii : np.ndarray (shape = [timesteps, agents])
+                        A numpy ndarray containing the radius 
+                        for each agent at each timestep
+                    subpoints: np.ndarray 
+                    (shape = [timesteps, agents, subpoints, 3]) (optional) 
+                        A numpy ndarray containing a list of subpoint position data 
+                        for each agent at each timestep. These values are 
+                        currently only used for fiber agents.
+                    plots : Dict[str, Any] (optional) 
+                        An object containing plot data already 
+                        in Simularium format
+
+        source_engine: str
+            A string specifying which simulation engine created these outputs. 
+            Current options:
+                'custom' : outputs are from an engine not specifically supported
+            Coming Soon:
+                'cytosim' : outputs are from CytoSim 
+                    (https://gitlab.com/f.nedelec/cytosim)
+                'readdy' : outputs are from ReaDDy 
+                    (https://readdy.github.io/)
+                'physicell' : outputs are from PhysiCell 
+                    (http://physicell.org/)
+        """
+        traj_reader_class = self._determine_trajectory_reader(source_engine)
+        self._data = traj_reader_class().read(data)
+
+    @staticmethod
+    def _determine_trajectory_reader(source_engine: str = "custom") -> Type[Reader]:
```

Comment:
I don't think you need the `Type` here... I don't _think_ :shrug:

Reply:
I copied how you guys did this in aicsimageio, but yes seems to be fine without it

Reply:
brb. searching for the random `Type[...]` call in `aicsimageio` lmao